app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -19,27 +19,25 @@ class BackupCodeSetupController < ApplicationController
     helper_method :in_multi_mfa_selection_flow?
 
     def index
-      generate_codes
       result = BackupCodeSetupForm.new(current_user).submit
       visit_result = result.to_h.merge(analytics_properties_for_visit)
```

Comment:
Calling out that the form analytics include `mfa_method_counts` and  `enabled_mfa_methods_count`. I _think_ the new logic here would produce the same expected output, since the codes weren't actually added to the account previously until `save_backup_codes` below?

Reply:
yep, exactly